resources/prosody-plugins/token/util.lib.lua

```diff
@@ -112,6 +112,14 @@ function Util:get_public_key(keyId)
     if content == nil then
         -- If the key is not found in the cache.
         module:log("debug", "Cache miss for key: "..keyId);
+        
+        local external_public_key = prosody.events.fire_event("pre-jitsi-authentication-public-key-fetch", keyId);
+        if external_public_key then 
+            module:log("debug", "External module loaded the key");
+            cache:set(keyId, external_public_key);
```

Comment:
This is the dangerous part, caching the key means in future any other JWT will be validated as acceptable for the general bucket.  This mixing of keys in a single cache isn't a good idea.  Instead if we want a cache for the plugin, it should manage its own cache from within its own module, and provide the key on the session from cache or URL.